generators/app/index.js

```diff
@@ -395,42 +415,45 @@ module.exports = class extends BaseGenerator {
-            ',\n      {\n        path: \'entity-audit\',\n            loadChildren: () => import(\'./entity-audit/entity-audit.module\').then(m => m.EntityAuditModule)\n      }'
-          );
-          // this.addAdminToModule('', 'EntityAudit', 'entity-audit', 'entity-audit', this.enableTranslation, this.clientFramework)
-        }
+        if (!this.skipClient) {
+          if (this.clientFramework === 'angularX') {
+            // add dependency required for displaying diffs
+            this.addNpmDependency('ng-diff-match-patch', '2.0.6');
+            // based on BaseGenerator.addAdminToModule
+            const adminModulePath = `${this.webappDir}app/admin/admin-routing.module.ts`;
+            this.rewriteFile(
+              adminModulePath,
+              'jhipster-needle-add-admin-route',
+              ',\n      {\n        path: \'entity-audit\',\n        loadChildren: () => import(\'./entity-audit/entity-audit.module\').then(m => m.EntityAuditModule)\n      }'
+            );
+            // this.addAdminToModule('', 'EntityAudit', 'entity-audit', 'entity-audit', this.enableTranslation, this.clientFramework)
+          }
 
-        // add new menu entry
-        this.addElementToAdminMenu('admin/entity-audit', 'list-alt', this.enableTranslation, this.clientFramework, 'entity-audit');
-        if (this.enableTranslation) {
-          this.languages.forEach((language) => {
-            let menuText = 'Entity Audit';
-            try {
-              menuText = JSON.parse(fs.readFileSync(`${this.templatePath()}/src/main/webapp/i18n/${language}/global.json`, 'utf8')).global.menu.admin['entity-audit'];
-            } catch (e) {
-              this.log('Cannot parse file');
-            }
-            this.addAdminElementTranslationKey('entity-audit', menuText, language);
-          });
+          // add new menu entry
+          this.addElementToAdminMenu('admin/entity-audit', 'list-alt', this.enableTranslation, this.clientFramework, 'entity-audit');
+          if (this.enableTranslation) {
+            this.languages.forEach((language) => {
+              let menuText = 'Entity Audit';
+              try {
+                menuText = JSON.parse(fs.readFileSync(`${this.templatePath()}/src/main/webapp/i18n/${language}/global.json`, 'utf8')).global.menu.admin['entity-audit'];
+              } catch (e) {
+                this.log('Cannot parse file');
+              }
+              this.addAdminElementTranslationKey('entity-audit', menuText, language);
+            });
+          }
         }
       },
 
       registering() {
         // Register this generator as a dev dependency
         this.addNpmDevDependency('generator-jhipster-entity-audit', packagejs.version);
-        // Register post-entity hook
+        // Register post-app and post-entity hook
         try {
+          this.registerModule('generator-jhipster-entity-audit', 'app', 'post', 'app', 'Add support for entity audit and audit log page');
```

Comment:
That's interesting. So it means that most module should register themselves as a post-app hook I guess.